docker/app/models/decoder.py

```
@@ -0,0 +1,177 @@
+            x: デコーダー入力 [batch_size, tgt_seq_len, d_model]
+            encoder_output: エンコーダー出力 [batch_size, src_seq_len, d_model]
+            tgt_mask: ターゲットマスク [batch_size, 1, tgt_seq_len, tgt_seq_len]
+            src_mask: ソースマスク [batch_size, 1, 1, src_seq_len]
+            cache: キャッシュ情報（推論時に使用）
+
+        Returns:
+            output: 出力テンソル [batch_size, tgt_seq_len, d_model]
+            new_cache: 更新されたキャッシュ情報
+        """
+        # キャッシュの初期化
+        new_cache = {} if cache is None else {}
```

Ruff (0.14.10)

Useless `if`-`else` condition

(RUF034)